### PR TITLE
Sample metadata pages

### PR DESCRIPTION
**Pull request by hanars**

### CODE REVIEW

File: ui/pages/Report/components/SampleMetadata.jsx [hunk 1]
```diff
@@ -4,13 +4,19 @@ import { connect } from 'react-redux'
...
 
+const FILENAME_LOOKUP = {
+  [ALL_PROJECTS_PATH]: 'All_AnVIL_Projects',
+  [GREGOR_PROJECT_PATH]: 'All_GREGoR_Projects',
+}
+
 const getDownloadFilename = (projectGuid, data) => {
-  const projectName = projectGuid && projectGuid !== 'all' && data.length && data[0].project_id.replace(/ /g, '_')
-  return `${projectName || 'All_AnVIL_Projects'}_${new Date().toISOString().slice(0, 10)}_Metadata`
+  const projectName = FILENAME_LOOKUP[projectGuid] || (data.length && data[0].project_id.replace(/ /g, '_'))
```
Comment (ShifaSZ) [inline ui/pages/Report/components/SampleMetadata.jsx:18]: The behavior changed slightly when `projectGuid` is None or '' the `projectName` changes from 'All_AnVIL_Projects' to be generated from the `data`.

**Review by ShifaSZ — CHANGES_REQUESTED** [review]
I have two minor questions.

File: seqr/views/apis/report_api_tests.py [hunk 2]
```diff
@@ -532,6 +541,22 @@ def test_sample_metadata_export(self, mock_google_authenticated):
         self.assertEqual(response.status_code, 200)
         self.assertDictEqual(response.json(), {'rows': []})
 
+        # Test all projects
+        responses.reset()
+        all_projects_url = reverse(sample_metadata_export, args=['all'])
+        response = self.client.get(all_projects_url)
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
```
[the remark this answers is not in the document]
Comment (hanars) [reply seqr/views/apis/report_api_tests.py:549]: the `responses.reset()` makes it so airtable would throw an error if called, so the fact that we have a 200 response means it wasn't called

File: ui/pages/Report/components/SampleMetadata.jsx (hunk 1 above)
Comment (hanars) [reply ui/pages/Report/components/SampleMetadata.jsx:18]: That case is only reachable before the user has actually loaded any data, so they would not be downloading the report anyways

File: seqr/views/apis/report_api_tests.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShifaSZ) [reply seqr/views/apis/report_api_tests.py:549]: Okay. I thought `responses.reset()` had the same behavior as `reset_mock()`, which doesn't clear the mocked value settings.

**Review by ShifaSZ — APPROVED** [review]

**Pull request merged**